assets/app/data/crowbar/steps.factory.js

```diff
@@ -4,72 +4,107 @@
         .module('suseData.crowbar')
         .factory('upgradeStepsFactory', upgradeStepsFactory);
 
-    upgradeStepsFactory.$inject = [];
+    upgradeStepsFactory.$inject = ['$state'];
     /* @ngInject */
-    function upgradeStepsFactory() {
-        var steps = initialSteps(),
-            factory = {
-                getAll: getSteps
-            };
+    function upgradeStepsFactory($state) {
+        var factory = {
+            steps: initialSteps(),
+            activeStep: {},
+            refeshStepsList: refeshStepsList,
+            setCurrentStepCompleted: setCurrentStepCompleted,
+            stepFinished: stepFinished
+        };
 
         return factory;
 
+        function stepFinished() {
+            return factory.activeStep.finished;
+        }
+        function setCurrentStepCompleted() {
+            factory.activeStep.finished = true;
+        }
+
+        function refeshStepsList() {
+            var currentState = $state.current.name,
+                isCompletedStep = true;
```

Comment:
This actually comes from richa's PR, not this one. Stupid gituhb doent allow linked patches properly :/

Also I think that var is not used anywhere for anything at all and was already in the code, I guess for preparation for tracking the step completition, but Im not sure I understand the reasoning of it at all. Should be removed in future patches as I dont think is useful.